src/test/java/com/rocketden/main/service/RoomServiceTests.java

```diff
@@ -0,0 +1,102 @@
+package com.rocketden.main.service;
+
+import com.rocketden.main.dao.RoomRepository;
+import com.rocketden.main.dto.room.CreateRoomRequest;
+import com.rocketden.main.dto.room.CreateRoomResponse;
+import com.rocketden.main.dto.room.JoinRoomRequest;
+import com.rocketden.main.dto.room.JoinRoomResponse;
+import com.rocketden.main.model.Room;
+import org.junit.Before;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+public class RoomServiceTests {
+
+    @Mock
+    private RoomRepository repository;
+
+    @Spy
+    @InjectMocks
+    private RoomService service;
+
+    @Before
+    public void setUp() {
+        service = new RoomService(repository);
+    }
+
+    @Test
+    public void createRoomSuccess() {
+        // Verify create room request succeeds and returns correct response
+        CreateRoomRequest request = new CreateRoomRequest();
+
+        // Mock generateRoomId to return a custom room id
+        Mockito.doReturn("012345").when(service).generateRoomId();
+        CreateRoomResponse response = service.createRoom(request);
+
+        verify(repository).save(Mockito.any());
```

Comment:
Syntax: this verifies that the `repository` object's `save` method was called once with any argument. 

See this article: https://www.baeldung.com/mockito-verify

Reply:
Could you make it specific to check that this is called on a `Room` object? If that's difficult no problem, just let me know.

Reply:
Just wanted to note that you left a 👍 on my comment, but didn't change this -- was that on purpose? If yes, sounds good as no change was required, but wanted to let you know in case.